File: tests/oopsla25/222.py
#!/usr/bin/env python3

import sys

def f222(s: str):
  assert s.index("b") == len(s)-1
  i = 0
  while i < len(s)-1:
    assert s[i] == "a"
    i += 1

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f222(my_string)

File: benchmarks/oopsla25/subjects/222.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f222(const char* s) {
    assert(strrchr(s, 'b') == s + strlen(s) - 1);
    int i = 0;
    while (i < strlen(s) - 1) {
        assert(s[i] == 'a');
        i++;
    }
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f222(my_string);
  return 0;
}
